layerstack/layer.py

```diff
@@ -416,17 +417,25 @@ class Layer(object):
         Directory from which to load the layer
     """
 
-    def __init__(self, layer_dir):
+    def __init__(self, layer_dir, model=None):
         self.layer_dir = layer_dir
+        self._model = model
```

Comment:
I don't think we actually want to save this here. I think we only want to use the model kwarg in the call to .args and .kwargs if _layer is a ModelLayerBase.